src/adapter/src/coord/command_handler.rs

```diff
@@ -821,11 +840,42 @@ impl Coordinator {
             // specified to be at `mz_now()` (which is usually the initial refresh)
             // (similarly to how we don't perform refreshes that were specified to be in the
             // past).
+
+            if acquire_read_holds {
+                self.acquire_read_holds_auto_cleanup(
+                    session,
+                    timestamp,
+                    &self.sufficient_collections_all_clusters(resolved_ids.0.iter()),
```

Comment:
Why is it correct to use all clusters here instead of the active cluster?

Reply:
Well, I didn't want to move or duplicate the cluster selection logic here, which currently happens [later](https://github.com/MaterializeInc/materialize/blob/e4cce43ce8b4fe54c57fe8edf3db94d929952cad/src/sql/src/plan/statement/ddl.rs#L2140-L2143).

I think it shouldn't cause any problems if we have read holds on a few more objects on some other clusters for a short while. But if this is too scary from a workload isolation perspective (e.g., optimizing the MV might take forever, at least for now that we don't have a timeout on optimizer calls), then maybe I can move or duplicate the cluster selection logic here, and then just acquire head holds from the relevant cluster. What do you think?

Reply:
I think it's worth doing that work to avoid possible isolation problems.

Reply:
Ok, then I'll factor out that logic into a function, and call it also here.

Reply:
Unfortunately, this is turning out more painful than expected, because `resolve_cluster` is a method of `StatementContext`, and we don't yet have a `StatementContext` in the purification.

It uses the `StatementContext` just to access the catalog, so I can probably hack this somehow, but I'm not really sure whether it's worth it at this point. E.g., maybe I can make a `resolve_cluster` variation that takes a `&'a dyn SessionCatalog` instead of a `StatementContext`.

Reply:
> maybe I can make a `resolve_cluster` variation that takes a `&'a dyn SessionCatalog`

Seems right to me. The StatementContext impl is just a wrapper around that anyway.

Reply:
Ok, done!